packages/server-wallet/src/wallet/__test__/fixtures/test-channel.ts

```diff
@@ -65,31 +73,31 @@ export class TestChannel {
    * Note - in cases where participants double-sign the same states, n might _not_
    * be the turnNum
    */
-  public state(n: number, bals?: [number, number], final = ''): State {
+  public state(n: number, bals?: [number, number]): State {
     return {
       ...this.channelConstants,
       appData: '0x',
-      isFinal: final === 'final',
-      turnNum: n,
+      isFinal: !!this.finalFrom && n >= this.finalFrom,
+      turnNum: Math.min(n, this.finalFrom || n),
```

Comment:
I think an explanatory comment is called for?

```suggestion
      turnNum: Math.min(n, this.finalFrom || n), // test channels adopt a countersigning strategy for final states, so the turn number doesn't progress after finalFrom.
```